src/components/redux/books/books.js

```diff
@@ -0,0 +1,73 @@
+/* eslint-disable import/no-extraneous-dependencies */
+import axios from 'axios';
+
+// Actions
+const ADD_BOOK = 'bookstore/books/ADD_BOOK';
+const REMOVE_BOOK = 'bookstore/books/REMOVE_BOOK';
+const GET_BOOKS = 'bookstore/books/GET_BOOKS';
+const api = 'https://us-central1-bookstore-api-e63c8.cloudfunctions.net/bookstoreApi/apps/v74KU1kb6ATX7KCQFVTz/books/';
+
+// initial State
+const initialState = [];
+
+// Reducer
+const booksReducer = (state = initialState, action) => {
+  switch (action.type) {
+    case ADD_BOOK:
+      return [...state, action.payload];
+    case REMOVE_BOOK:
+      return state.filter((book) => book.id !== action.id);
+    case GET_BOOKS:
+      return action.payload;
+    default:
+      return state;
+  }
+};
+
+// Action Creators
+const getBooks = () => async (dispatch) => {
+  const allBooks = [];
+  const response = await axios.get(api);
```

Comment:
- [ ] As required, I kindly suggest that you make use of `createAsyncThunk` to fetch data from api.

## EXAMPLE

```
export const fetchBooksData = createAsyncThunk('fetchBooksData', async () => {
  const response = await fetch(api);
  const data = await response.json();
  const booksList = [];
  Object.keys(data).forEach((id) => {
    booksList.push({
      item_id: id,
      title: data[id][0].title,
      author: data[id][0].author,
      category: data[id][0].category,
    });
  });
  return booksList;
});
```


The above code is just an example and may not work directly for your code.

Reply:
Hi Reviewer, 
I have added createAsyncThunk as you suggested.

Reply:
> Hi Reviewer, I have added createAsyncThunk as you suggested.

@Mekuseo, that's great